service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/azure/managedIdentity/GetWorkspaceManagedIdentityStepTest.java

```diff
@@ -0,0 +1,146 @@
+import com.azure.core.http.HttpResponse;
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.msi.MsiManager;
+import com.azure.resourcemanager.msi.models.Identities;
+import com.azure.resourcemanager.msi.models.Identity;
+import java.util.UUID;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoSession;
+import org.mockito.quality.Strictness;
+import org.springframework.http.HttpStatus;
+
+@Tag("azure-unit")
+public class GetWorkspaceManagedIdentityStepTest {
+  private MockitoSession mockito;
+  @Mock private FlightContext mockFlightContext;
+  @Mock private FlightMap mockWorkingMap;
+  @Mock private AzureCloudContext mockAzureCloudContext;
+  @Mock private CrlService mockCrlService;
+  @Mock private MsiManager mockMsiManager;
+  @Mock private Identities mockIdentities;
+  @Mock private AzureConfiguration mockAzureConfig;
+  @Mock private HttpResponse mockHttpResponse;
+  @Mock private ResourceDao mockResourceDao;
+  @Mock private Identity mockIdentity;
+
+  @BeforeEach
+  public void setup() {
+    // initialize session to start mocking
+    mockito =
+        Mockito.mockitoSession().initMocks(this).strictness(Strictness.STRICT_STUBS).startMocking();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    mockito.finishMocking();
+  }
+
+  @Test
+  void testSuccess() throws InterruptedException {
+    var workspaceId = UUID.randomUUID();
+    var creationParameters =
+        ControlledAzureResourceFixtures.getAzureManagedIdentityCreationParameters();
+    var identityResource =
+        ControlledAzureResourceFixtures.makeDefaultControlledAzureManagedIdentityResourceBuilder(
+                creationParameters, workspaceId)
+            .build();
+
+    createMockFlightContext();
+
+    when(mockCrlService.getMsiManager(any(), any())).thenReturn(mockMsiManager);
+    when(mockMsiManager.identities()).thenReturn(mockIdentities);
+    when(mockIdentities.getByResourceGroup(
+            mockAzureCloudContext.getAzureResourceGroupId(),
+            identityResource.getManagedIdentityName()))
+        .thenReturn(mockIdentity);
+    when(mockResourceDao.getResource(workspaceId, identityResource.getResourceId()))
+        .thenReturn(identityResource);
+    when(mockIdentity.name()).thenReturn(UUID.randomUUID().toString());
+    when(mockIdentity.principalId()).thenReturn(UUID.randomUUID().toString());
+    when(mockIdentity.clientId()).thenReturn(UUID.randomUUID().toString());
+
+    var step =
+        new GetWorkspaceManagedIdentityStep(
+            mockAzureConfig,
+            mockCrlService,
+            workspaceId,
+            mockResourceDao,
+            identityResource.getResourceId());
+    assertThat(step.doStep(mockFlightContext), equalTo(StepResult.getStepResultSuccess()));
```

Comment:
Are there any useful assertions to be made around
1) the output parameters of the flight?
2) the proper calls are made to the sdk methods?

Same question for GetPetManagedIdentityStepTest